mpas_analysis/ocean/streamfunction_moc.py

```diff
@@ -345,34 +364,48 @@ def _compute_moc_climo_postprocess(self):  # {{{
 
         config = self.config
 
-        dvEdge, areaCell, refBottomDepth, latCell, nVertLevels, \
-            refTopDepth, refLayerThickness = self._load_mesh()
-
         self.regionNames = config.getExpression(self.sectionName,
                                                 'regionNames')
 
         # Load basin region related variables and save them to dictionary
+        mpasMeshName = config.get('input', 'mpasMeshName')
+        regionMaskDirectory = config.get('regions', 'regionMaskDirectory')
+
+        regionMaskFile = '{}/{}_MOCBasinsAndTransectMasks.nc'.format(
+                regionMaskDirectory, mpasMeshName)
+
+        if not os.path.exists(regionMaskFile):
+            raise IOError('Regional masking file {} for MOC calculation '
+                          'does not exist'.format(regionMaskFile))
         # NB: The following will need to change with new regional mapping files
-        regionMaskFiles = config.get(self.sectionName, 'regionMaskFiles')
-        if not os.path.exists(regionMaskFiles):
+        regionMaskFile = config.get(self.sectionName, 'regionMaskFile')
+        if not os.path.exists(regionMaskFile):
             raise IOError('Regional masking file for MOC calculation '
                           'does not exist')
-        iRegion = 0
         self.dictRegion = {}
+        dsMask = xr.open_dataset(regionMaskFile)
```

Comment:
I open the mask/transect file only once outside the loop over regions.  I think this makes more sense.